src/core/mac/mac.cpp

```diff
@@ -1269,6 +1269,14 @@ void Mac::HandleTransmitDone(TxFrame &aFrame, RxFrame *aAckFrame, otError aError
 
             mBroadcastTransmitCount = 0;
         }
+
+#if OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2
+        // Verify Enh-ACK integrity by checking its MIC
+        if ((aError == OT_ERROR_NONE) && (aAckFrame != nullptr))
+        {
+            aError = (ProcessEnhAckSecurity(aFrame, *aAckFrame) == OT_ERROR_NONE) ? OT_ERROR_NONE : OT_ERROR_NO_ACK;
```

Comment:
If a malformed enh-ack is injected before a genuine enh-ack, the radio state machine will jump to some state like tx frame acked. So the following genuine enh-ack will just be ignored. Also, this case will not happen because ack sending does not go through CSMA/CA process. For example, A is the data frame transmitter, B is the intended genuine receiver, C is an intruder. When A sends out a data frame. B and C both receives this. B will send out genuine enh-ack after AIFS(192us) without doing any CCA. If C sends out a malformed ack, this ack will collide with B's ack and results in an NO_ACK for A.

Reply:
So if there'an intruder, then A couldn't get the genuine enh-ack, right?

Reply:
Yes

Reply:
Ok, make sense to me.